### PR TITLE
Add extra support for more getters for the Validator class

### PR DESCRIPTION
**Pull request by marcin-cb**
### What changed? Why?
Adding support for more getters to expose the data behind the Validator object.

### Example
`validator.toString()`
```
'Validator { Id: 0x921018e49cf63a8573ea2e0c665d025325361b417233d6727907b205cf2c420e153060297b60dcbf6d42871cf8610d2f, Status: active, Exit Epoch: 18446744073709551615, Network ID: ethereum-holesky, Asset ID: ETH, Index: 1768213, Public Key: 0x921018e49cf63a8573ea2e0c665d025325361b417233d6727907b205cf2c420e153060297b60dcbf6d42871cf8610d2f, Slashed: false, Withdrawable Epoch: 18446744073709551615, Withdrawal Address: 0x0100000000000000000000008d082f80606b8f93827a8a8e8c1c48950be60a33, Effective Balance: { Amount: 32000000000000000000, AssetID: { eth, Decimals: 18, NetworkID: ethereum-holesky } }, Balance: { Amount 32000113927000000000, Asset: { AssetID: eth, Decimals: 18, NetworkID: ethereum-holesky } }, Activation Epoch: 67501 }'
```
`validator.toJSON()`
```
'{"asset_id":"ETH","details":{"activationEpoch":"67501","balance":{"amount":"32000113927000000000","asset":{"asset_id":"eth","decimals":18,"network_id":"ethereum-holesky"}},"effective_balance":{"amount":"32000000000000000000","asset":{"asset_id":"eth","decimals":18,"network_id":"ethereum-holesky"}},"exitEpoch":"18446744073709551615","index":"1768213","public_key":"0x921018e49cf63a8573ea2e0c665d025325361b417233d6727907b205cf2c420e153060297b60dcbf6d42871cf8610d2f","slashed":false,"withdrawableEpoch":"18446744073709551615","withdrawal_address":"0x0100000000000000000000008d082f80606b8f93827a8a8e8c1c48950be60a33"},"network_id":"ethereum-holesky","status":"active","validator_id":"0x921018e49cf63a8573ea2e0c665d025325361b417233d6727907b205cf2c420e153060297b60dcbf6d42871cf8610d2f"}'
```

### CODE REVIEW

File: src/coinbase/validator.ts [hunk 1]
```diff
@@ -151,11 +156,119 @@ export class Validator {
...
+  }
+
+  /**
+   * Returns the string representation of the Validator including all its details.
+   *
+   * @returns The string representation of the Validator including all its details.
    */
   public toString(): string {
-    return `Id: ${this.getValidatorId()} Status: ${this.getStatus()}`;
+    return `Validator { Id: ${this.getValidatorId()}, Status: ${this.getStatus()}, Exit Epoch: ${this.getExitEpoch()}, Network ID: ${this.getNetworkId()}, Asset ID: ${this.getAssetId()}, Index: ${this.getIndex()}, Public Key: ${this.getPublicKey()}, Slashed: ${this.isSlashed()}, Withdrawable Epoch: ${this.getWithdrawableEpoch()}, Withdrawal Address: ${this.getWithdrawalAddress()}, Effective Balance: { Amount: ${this.getEffectiveBalance()?.amount}, AssetID: { ${this.getEffectiveBalance()?.asset.asset_id}, Decimals: ${this.getEffectiveBalance()?.asset.decimals}, NetworkID: ${this.getEffectiveBalance()?.asset.network_id} } }, Balance: { Amount ${this.getBalance()?.amount}, Asset: { AssetID: ${this.getBalance()?.asset.asset_id}, Decimals: ${this.getBalance()?.asset.decimals}, NetworkID: ${this.getBalance()?.asset.network_id} } }, Activation Epoch: ${this.getActivationEpoch()} }`;
```
Comment (drohit-cb) [inline src/coinbase/validator.ts:263]: I think toString should still be the simpler one. Primarily used of logging etc. This will be too much to log.

Comment (marcin-cb) [reply src/coinbase/validator.ts:263]: Yeah I see. Truly wasn't sure what to do here. I'm fine with simplifying this.
Are you suggesting just printing out the top level data values? or what subset specifically would be important?

File: src/coinbase/validator.ts [hunk 2]
```diff
@@ -151,11 +156,119 @@ export class Validator {
...
+  public getAssetId(): string {
+    return this.model.asset_id;
+  }
+
+  /**
+   * Returns the activation epoch of the validator.
+   *
+   * @returns The activation epoch as a string.
+   */
+  public getActivationEpoch(): string {
```
Comment (drohit-cb) [inline src/coinbase/validator.ts:181]: the problem with this is - these are very ethereum specific.

In future if we say support sol validators, do we still want user to do `validator. getActivationEpoch()` ?

Do we want to support a helper function of this format instead `validator. getEthereumDetails().getActivationEpoch()` ?

Comment (marcin-cb) [reply src/coinbase/validator.ts:181]: I'd be more in favor of revisiting this when we cross that bridge. i.e. I don't see us exposing Solana validator staking anytime soon.

File: src/coinbase/validator.ts (hunk 1 above)
Comment (drohit-cb) [reply src/coinbase/validator.ts:263]: Just id and status like before.

**Review by drohit-cb — APPROVED** [review]

**Pull request merged**